conftest.py

```diff
@@ -29,46 +31,44 @@ def pytest_configure(config):
     reporter = FixtureReporter(config)
     config.pluginmanager.register(reporter, 'fixturereporter')
 
+def _max_fname_len(file_system_path, default_max_len=80):
+    name_max_str = [s for s in os.pathconf_names if s.endswith('_NAME_MAX')]
```

Comment:
_PATH_MAX?

Reply:
this is for a (prefix of) a path component, not full path

Reply:
turning into iterators is for specifying cleanup actions after fixture goes out of scope